agent/structs/structs.go

```diff
@@ -1464,6 +1463,16 @@ func (s *NodeService) Validate() error {
 					"A SidecarService cannot have a nested SidecarService"))
 			}
 		}
+
+		if s.Connect.SidecarService.Port == 0 {
```

Comment:
We will need to update the docs for this, yes? https://www.consul.io/docs/connect/registration/sidecar-service#minimal-example

Reply:
Actually, I moved the check for the side car service only for the catalog endpoint. The agent endpoint could still continue to use it. So might not need to update the docs?